File: add-env-prod.js
const https = require('https');

const projectId = 'esenciadebelleza';
const teamId = 'tomassette-progs-projects';
const variableName = 'SUPABASE_SERVICE_ROLE_KEY';
const variableValue = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InlqYW5vYnNmemN3cHVzeW52bHVuIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4MTY5NzE3OCwiZXhwIjoyMDk3MjczMTc4fQ.Ph7vXLcfU4Dv3YJ3jzcIOnehn9v_x0FvDMjtyDfkse8';
const token = process.env.VERCEL_TOKEN;

if (!token) {
  console.error('❌ VERCEL_TOKEN not set');
  process.exit(1);
}

const data = JSON.stringify({
  key: variableName,
  value: variableValue,
  target: ['production'],
});

const options = {
  hostname: 'api.vercel.com',
  path: `/v8/projects/${projectId}/env?teamId=${teamId}`,
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length,
    'Authorization': `Bearer ${token}`,
  },
};

const req = https.request(options, (res) => {
  let body = '';
  res.on('data', (chunk) => { body += chunk; });
  res.on('end', () => {
    if (res.statusCode === 200 || res.statusCode === 201) {
      console.log('✅ Variable agregada a Production');
      console.log(body);
    } else {
      console.error(`❌ Error ${res.statusCode}:`);
      console.error(body);
    }
  });
});

req.on('error', (e) => {
  console.error('❌ Error:', e);
  process.exit(1);
});

req.write(data);
req.end();

File: add-env-prod.py
#!/usr/bin/env python3
import os
import subprocess
import json

# Variables
PROJECT = "esenciadebelleza"
TEAM = "tomassette-progs-projects"
VAR_NAME = "SUPABASE_SERVICE_ROLE_KEY"
VAR_VALUE = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InlqYW5vYnNmemN3cHVzeW52bHVuIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4MTY5NzE3OCwiZXhwIjoyMDk3MjczMTc4fQ.Ph7vXLcfU4Dv3YJ3jzcIOnehn9v_x0FvDMjtyDfkse8"

print(f"✅ Agregando {VAR_NAME} a Vercel Production")
print(f"   Proyecto: {PROJECT}")
print(f"   Team: {TEAM}")
print(f"   Valor: {VAR_VALUE[:30]}...")

cmd = [
    "powershell", "-NoProfile", "-Command",
    f"""
    $ErrorActionPreference = 'Stop'
    
    # Obtener el token
    $lines = @(Get-Content .env.local)
    $oidcToken = ($lines | Where-Object {{ $_ -match 'VERCEL_OIDC_TOKEN' }} | Select-Object -First 1).Split('=')[1].Trim('\"')
    
    if (-not $oidcToken) {{
        Write-Error "No VERCEL_OIDC_TOKEN found in .env.local"
        exit 1
    }}
    
    Write-Host "Token encontrado, intentando agregar variable..."
    
    # Usar curl para agregar la variable (simulando la solicitud)
    $payload = @{{
        key = '{VAR_NAME}'
        value = '{VAR_VALUE}'
        target = @('production')
    }} | ConvertTo-Json
    
    Write-Host "Payload: $payload"
    """
]

try:
    result = subprocess.run(cmd, capture_output=True, text=True)
    print(result.stdout)
    if result.stderr:
        print("STDERR:", result.stderr)
    if result.returncode != 0:
        print(f"Error (exit code {result.returncode})")
except Exception as e:
    print(f"Error ejecutando script: {e}")
